includes/alarmHandler.py

```diff
@@ -36,20 +37,20 @@ def processAlarmHandler(typ, freq, data):
 	@return:    nothing
 	@exception: Exception if starting a Thread failed
 	"""
+	#copy objects to avoid issues if the objects will be changed by plugin's or during asynch/threaded processing 
+	dctyp = deepcopy(typ)
```

Comment:
Ich glaube wenn wir an dieser Stelle kopieren, besteht das Problem weiterhin.
Denn alle Plugins greifen ja auf EXAKT die selbe Kopie zu - oder täusche ich mich?

Reply:
Muss ich mir morgen mal ansehen.

Reply:
zumindest an dieser Stelle bringt es nichts wenn wir verhindern wollen, dass die Plugins sich gegenseitig die Daten verändern könnten.

Passenderweise müsste es dann wohl in die Schleife, die die Plugins aufruft, sodass jeder Plugin Aufruf eine eigene Kopie der Daten erhält.

Reply:
Guter Punkt. Habe das mal umgebaut. Ich muss das aber noch auf meinem System testen.

Reply:
@f-kessler Läuft stabil?

Reply:
Irgendwas läuft da noch nicht richtig. Die Kopien werden offenbar auch überschrieben. Zumindest landen in der Datenbank viele Alarme mit der selben RIC. Ich habe zwei Vermutungen. Entweder es liegt daran, dass in Zeile 44 bereits der Thread erzeugt wird, aber noch mit original Objekten an dieser Stelle gearbeitet wird. Oder es liegt daran, dass die Kopien immer den selben Namen haben und das Kopieren die Kopie überschreibt.
Hast du vielleicht eine Idee dazu?

Reply:
So wie sich mir das darstellt, besteht dieses Problem nur beim data-Dict.
> Schlimmer ist es jedoch, wenn Seiteneffekte auftreten, mit denen man nicht gerechnet hat, und dadurch Fehler verursacht werden. Dies kann auch in Python passieren. Dies kann dann geschehen, wenn Listen oder Dictionaries als Parameter übergeben werden. 

Die anderen DeepCopys sind also vermutlich gar nicht notwendig.
Evtl reicht es schon das deepCopy direkt in den Funktionsaufruf zu setzen:
`plugin.run(typ, freq, deepcopy(data))` ?

Reply:
Das ist eine gute Idee. Werde das mal am Wochenende bei mir umschreiben. Hatte inzwischen auch gute Ergebnisse mit einem deepcopy vor dem Aufruf des Threads und einem weiteren deepcopy in der for-Schleife erzielt. Dein Vorschlag ist aber deutlich eleganter als stumpf neue Objekte anzulegen.

Reply:
Bin leider erst neulich dazu gekommen deinen Vorschlag zu testen. Bin mit dem Ergebnis sehr zufrieden.